### PR TITLE
EL-2562: Chore: add tests to improve code coverage

### PR DESCRIPTION
**Pull request by sryalls**
[EL-2562](https://dsdmoj.atlassian.net/browse/EL-2562)

This commit introduces general test infrastructure improvements as a pre-requisite to enable EL-2457-client-date-of-birth-editing branch to pass the 67% code coverage gate.

## Background
The DOB editing branch (EL-2457-client-date-of-birth-editing) requires 67% function coverage to pass CI. Rather than mixing general infrastructure improvements with feature-specific code, this separate branch provides more general test improvements.

## Changes Added
- Form validation tests (formValidationRules.spec.ts, validateForm.spec.ts)
- Controller test coverage for caseDetailsController, editClientDetailsController, yourCasesController
- TypeScript CSRF type definitions to resolve import compilation errors
- Tests cover edge cases, error scenarios, and return structure validation

Before you ask people to review this PR:

- [x] **Tests and linting** are passing.  
- [x] **Branch is up to date** with `main` (no merge conflicts).  
- [x] **No unnecessary whitespace changes** (avoid unnecessary diffs).  
- [x] **PR description clearly explains** what changed and why, with a JIRA ticket/Trello link.  
- [x] **Diff has been checked** for any unexpected changes.  
- [x] **Commit messages are clear** and explain what will change, if individual commit, needs to be revisited retroactively.

[EL-2562]: https://dsdmoj.atlassian.net/browse/EL-2562?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/unit/helpers/validateForm.spec.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        };
+        
+        const result = validateForm(fields);
+        
+        expect(result.formIsInvalid).to.be.false;
+        expect(result.inputErrors).to.be.empty;
+        expect(result.errorSummaryList).to.be.empty;
+      });
+
+      it('should return no errors for valid combined form', () => {
```
[the remark this answers is not in the document]
Comment (MazOneTwoOne) [reply tests/unit/helpers/validateForm.spec.ts:36]: That makes sense now.

File: tests/unit/helpers/formValidationRules.spec.ts [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        expect(formatError?.inputError?.fieldName).to.equal('emailAddress');
+      });
+
+      it('should validate various email formats correctly', () => {
+        const validEmails = [
+          'test@example.com',
+          'user.name@domain.co.uk',
+          'user+tag@example.org',
+          'user123@test-domain.com',
+          'user..name@domain.com' // consecutive dots are allowed by this simple regex
```
[the remark this answers is not in the document]
Comment (sryalls) [reply tests/unit/helpers/formValidationRules.spec.ts:159]: you mean, should we fix the regex not to allow consecutive dots? Yes absolutely. Even better, use a lib like https://www.npmjs.com/package/email-validator because who wants to deal with regex if they don't have to?

[the remark this answers is not in the document]
Comment (MazOneTwoOne) [reply tests/unit/helpers/formValidationRules.spec.ts:159]: Cool - I will create a ticket for that.

File: tests/unit/helpers/validateForm.spec.ts [hunk 3]
```diff
@@ -0,0 +1,239 @@
+import { describe, it } from 'mocha';
```
[the remark this answers is not in the document]
Comment (sryalls) [reply tests/unit/helpers/validateForm.spec.ts:16]: Description added

File: tests/unit/helpers/formValidationRules.spec.ts [hunk 4]
```diff
@@ -0,0 +1,305 @@
+import { describe, it } from 'mocha';
```
[the remark this answers is not in the document]
Comment (sryalls) [reply tests/unit/helpers/formValidationRules.spec.ts:16]: Description added
